src/api.js

```diff
@@ -0,0 +1,21 @@
+const BASE_URL = "https://bootcamp-api.codeit.kr";
+
+export async function getUser() {
+  try {
+    const response = await fetch(`${BASE_URL}/api/sample/user`);
+    const user = await response.json();
+    return user;
+  } catch (error) {
+    console.log("유저 정보 불러오기 실패");
+  }
+}
```

Comment:
fetch를 사용하는 경우에는 개발자가 직접 fetch가 정상적으로 실행됐는지 확인하는 코드가 필요한데요~ [참고](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#checking_that_the_fetch_was_successful)

이렇게 ok라는 값을 이용해서 확인을 해주고 만약 ok값이 false일 경우에는 에러를 던져줘야해요~

```suggestion
export async function getUser() {
  try {
    const response = await fetch(`${BASE_URL}/api/sample/user`);
    if(!response.ok) {
      throw new Error('유저 데이터를 불러올 수 없습니다.');
    }
    const user = await response.json();
    return user;
  } catch (error) {
    console.log("유저 정보 불러오기 실패");
  }
}
```

Reply:
추가적으로 흐름은 잘 작성해주셨는데요! 에러처리에 대해서 고민해볼 필요가 있을 것 같아요~ 
사실 저도 에러처리 부분이 처음 공부할때 엄청 난해하고 이해가 안되는 부분이었는데요!

우리의 서비스를 이용하는 유저 입장에서 생각했을 때 만약 서비스내의 기능이 제대로 동작하지 않으면 사용자가 에러가 발생했다는 상황을 인지할 수 있어야 하는데요~ 사용자에게 무언가를 보여주는 영역은 컴포넌트일 것 같아요! 즉, 에러에 대한 처리는 컴포넌트(계층으로 얘기한다면 View 혹은 UI)에서 해줘야 해요!

현재 작성되어 있는 코드를 보면 response.ok의 값이 false일 때 에러를 발생시켜서, 이 에러를 catch문에서 확인해서 콘솔에 에러를 찍고 null을 리턴하는데요! 이렇게 되면 에러가 컴포넌트까지 전달되지 않는다는 문제가 있을 것 같아요~ 따라서 에러를 컴포넌트까지 전달을 해줘야하고 컴포넌트에서 에러를 잡아서(catch) 에러 처리를 해줘야해요!

코드로 작성을 해보면 좋을 것 같은데요~
api 함수에서는 이렇게만 작성을 해주시고나서
```js
export const getUser = () => {
  const response = await fetch(`${API_URL}/api/sample/user`);
  
  if (!response.ok) {
    throw new Error('Failed to fetch profile data');
  }

  return await response.json();
}
```

이렇게 컴포넌트에서 api 함수를 try-catch로 감싸서 실행을 한 후에, api함수에서 에러가 발생했다면 catch 블록 내부의 코드가 실행이 되고 여기서 상황에 맞는 에러처리를 해주면 돼요! 지금 당장 공부를 하는 시점에서는 console.error로 출력해주는것도 괜찮다고 생각은 하고 window.alert를 이용해주는 것도 괜찮을 것 같아요!
```js
export default function Component () {
  const [data, setData] = useState(null);

  useEffect(() => {
    const fetchUser = async () => {
      try {
        const result = await getUser();
        setData(result);
      } catch(error) {
        // 1. 팝업을 띄운다.
        // 2. 화면에 에러 메시지를 보여준다.
        // 3. 그 외 서비스 정책에 맞게 에러처리를 한다.
      }
    }
    
    fetchUser();
  }, [])
}
```

Reply:
멘토님이 설명해주신대로 코드를 수정해보니 에러처리를 조금 어색하게 했던 것 같습니다 ㅎㅎㅎ 정말 감사합니다